server/game/drawcard.js

```diff
@@ -58,9 +61,32 @@ class DrawCard extends BaseCard {
         clone.strengthSet = this.strengthSet;
         clone.tokens = Object.assign({}, this.tokens);
         clone.traits = this.traits.clone();
+
         return clone;
     }
 
+    setupDuplicateAbility(ability) {
+        let dupeCondition = event => event.card === this.parent && this.parent.canBeSaved() && event.allowSave && this.parent.owner.promptDupes;
+
+        this.interrupt({
+            canCancel: true,
```

Comment:
Forgot one thing in my review - this interrupt also needs `cannotBeCanceled: true` to prevent things like Treachery or Fear Cuts Deeper Than Swords from triggering inappropriately.